app/org/sagebionetworks/bridge/validators/SurveyValidator.java

```diff
@@ -132,11 +132,24 @@ private void validateRules(Errors errors, List<SurveyElement> elements) {
             SurveyElement element = elements.get(i);
             if (element instanceof SurveyQuestion) {
                 for (SurveyRule rule : ((SurveyQuestion)element).getConstraints().getRules()) {
-                    if (alreadySeenIdentifiers.contains(rule.getSkipToTarget())) {
-                        errors.pushNestedPath("elements["+i+"]");
+                    errors.pushNestedPath("elements["+i+"]");
+                    // Validate the rule either has a skipTo target, or an endSurvey = TRUE, but not both.
+                    if (rule.getSkipToTarget() != null && rule.getEndSurvey() != null) {
+                        rejectField(errors, "rule", "cannot have a skipTo target and an endSurvey property");
+                    } 
+                    // But must have either a skipTo target or an endSurvey property
+                    else if (rule.getSkipToTarget() == null && rule.getEndSurvey() == null) {
+                        rejectField(errors, "rule", "must have a skipTo target or an endSurvey property");
+                    }
+                    // Validate the endSurvey value cannot be false
+                    else if (Boolean.FALSE.equals(rule.getEndSurvey())) {
+                        rejectField(errors, "rule", "cannot set endSurvey to false");
+                    } 
```

Comment:
As implemented in SurveyRule, it's impossible for SurveyRule.getEndSurvey() to be false. Do we still need this check here? If we're worried that a future change might accidentally make getEndSurvey()==false to be possible, would it be better to just allow it in SurveyRule and always validate it here? This way, we have the logic in a single place and can be easily tested.